### PR TITLE
HCAL Physim pedestal2015

### PR DESCRIPTION
**Pull request by kodolova**
Add configuration for NZS AlcaReco producer over TestEnabledEcalHcalDT dataset

### CODE REVIEW

**Comment by diguida** [comment]
please test

**Comment by diguida** [comment]
@kodolova 
With this PR you are adding only the definition of the AlCaReco. If you want to run this in central sequences, you should add this to `Configuration/StandardSequences/python/AlCaRecoStream_Specials_cff.py`

**Comment by diguida** [comment]
@kodolova you should also add the output module definition into `Configuration/EventContent/python/AlCaRecoOutput_cff.py`

**Comment by mmusich** [comment]
@kodolova @diguida (mostly out of curiosity) from [here](https://github.com/cms-sw/cmssw/blob/CMSSW_7_5_X/Configuration/AlCa/python/autoAlca.py#L17) looks like already `HcalCalMinBias` was run on `HcalNZS` AlCaRAW in Run1.
This PR introduces `ALCARECOHcalCalPedestal` which is different form `ALCARECOHcalCalMinBias` for the `hcalLocalRecoNZS` local reconstruction. Is the former one with standard local reco still meant to be used? Thanks

**Comment by kodolova** [comment]
ALCARECOHcalCalPedestal should be run only on TestEnabledEcalHcalDT dataset **instead of ALCARECOHcalCalMinBias**. It should not be run on anything else. In turn ALCARECOHcalCalMinBias **should not run on TestEnabledEcalHcalDT**. 
(!) hcalLocalRecoNZS is needed for the StandardSequence (!)
My questions are:
  - What is the sequence that run over TestEnabledEcalHcalDT? Does it include hcalLocalRecoNZS?
     If yes I will remove hbherecoMB, hfrecoMB, horecoMB instances from ALCARECOHcalCalPedestal

File: Calibration/HcalAlCaRecoProducers/python/ALCARECOHcalCalPedestal_cff.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import FWCore.ParameterSet.Config as cms
+
+#------------------------------------------------
+#AlCaReco filtering for HCAL minbias:
+#------------------------------------------------
+
+from Calibration.HcalAlCaRecoProducers.ALCARECOHcalCalMinBias_cff import *
+hcalDigiAlCaMB.InputLabel = 'hltHcalCalibrationRaw'
+from EventFilter.HcalRawToDigi.HcalRawToDigi_cfi import *
+#hcalDigis.InputLabel = 'hltHcalCalibrationRaw'
```
Comment (diguida) [inline Calibration/HcalAlCaRecoProducers/python/ALCARECOHcalCalPedestal_cff.py]: @kodolova 
Removing the definition of `hcalDigis.InputLabel` breaks the standard sequence definition:

````
  File "/afs/cern.ch/cms/sw/ReleaseCandidates/vol0/slc6_amd64_gcc491/cms/cmssw-patch/CMSSW_7_5_X_2015-03-18-1400/bin/slc6_amd64_gcc491/cmsDriver.py", line 43, in <module>
    run()
  File "/afs/cern.ch/cms/sw/ReleaseCandidates/vol0/slc6_amd64_gcc491/cms/cmssw-patch/CMSSW_7_5_X_2015-03-18-1400/bin/slc6_amd64_gcc491/cmsDriver.py", line 15, in run
    configBuilder.prepare()
  File "/afs/cern.ch/cms/sw/ReleaseCandidates/vol0/slc6_amd64_gcc491/cms/cmssw-patch/CMSSW_7_5_X_2015-03-18-1400/python/Configuration/Applications/ConfigBuilder.py", line 2084, in prepare
    self.addStandardSequences()
  File "/afs/cern.ch/cms/sw/ReleaseCandidates/vol0/slc6_amd64_gcc491/cms/cmssw-patch/CMSSW_7_5_X_2015-03-18-1400/python/Configuration/Applications/ConfigBuilder.py", line 717, in addStandardSequences
    getattr(self,"prepare_"+stepName)(sequence = '+'.join(stepSpec))
  File "/afs/cern.ch/cms/sw/ReleaseCandidates/vol0/slc6_amd64_gcc491/cms/cmssw-patch/CMSSW_7_5_X_2015-03-18-1400/python/Configuration/Applications/ConfigBuilder.py", line 1228, in prepare_ALCA
    alcaConfig=self.loadDefaultOrSpecifiedCFF(sequence,self.ALCADefaultCFF)
  File "/afs/cern.ch/cms/sw/ReleaseCandidates/vol0/slc6_amd64_gcc491/cms/cmssw-patch/CMSSW_7_5_X_2015-03-18-1400/python/Configuration/Applications/ConfigBuilder.py", line 1181, in loadDefaultOrSpecifiedCFF
    l=self.loadAndRemember(defaultCFF,unsch)
  File "/afs/cern.ch/cms/sw/ReleaseCandidates/vol0/slc6_amd64_gcc491/cms/cmssw-patch/CMSSW_7_5_X_2015-03-18-1400/python/Configuration/Applications/ConfigBuilder.py", line 290, in loadAndRemember
    self.process.load(includeFile)
  File "/afs/cern.ch/cms/sw/ReleaseCandidates/vol0/slc6_amd64_gcc491/cms/cmssw-patch/CMSSW_7_5_X_2015-03-18-1400/python/FWCore/ParameterSet/Config.py", line 515, in load
    self.extend(sys.modules[moduleName])
  File "/afs/cern.ch/cms/sw/ReleaseCandidates/vol0/slc6_amd64_gcc491/cms/cmssw-patch/CMSSW_7_5_X_2015-03-18-1400/python/FWCore/ParameterSet/Config.py", line 532, in extend
    self.__setattr__(name,item)
  File "/afs/cern.ch/cms/sw/ReleaseCandidates/vol0/slc6_amd64_gcc491/cms/cmssw-patch/CMSSW_7_5_X_2015-03-18-1400/python/FWCore/ParameterSet/Config.py", line 364, in __setattr__
    raise ValueError(msg1+s.label_()+msg2)
ValueError: Trying to override definition of hcalDigis while it is used by the sequence RawToDigi
 new object defined in: /afs/cern.ch/cms/sw/ReleaseCandidates/vol0/slc6_amd64_gcc491/cms/cmssw-patch/CMSSW_7_5_X_2015-03-18-1400/python/EventFilter/HcalRawToDigi/HcalRawToDigi_cfi.py
 existing object defined in: /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_5_X_2015-03-18-1400/python/Configuration/StandardSequences/RawToDigi_cff.py
````

**Comment by diguida** [comment]
@kodolova 
please see https://github.com/cms-sw/cmssw/pull/8350#discussion_r26903997
Please have a look at the `HcalRawToDigi` fragment.

File: Configuration/StandardSequences/python/AlCaRecoStream_Specials_cff.py [hunk 2]
```diff
@@ -82,6 +82,24 @@
...
+from Calibration.HcalAlCaRecoProducers.ALCARECOHcalCalPedestal_cff import *
+
+from DQMOffline.Configuration.AlCaRecoDQM_cff import *
+
+pathALCARECOHcalCalPedestal = cms.Path(seqALCARECOHcalCalPedestal*ALCARECOHcalCalPhisymDQM)
+
+from Configuration.EventContent.AlCaRecoOutput_cff import *
+
+ALCARECOStreamHcalCalPedestal = cms.FilteredStream(
+        responsible = 'Grigory Safronov',
```
Comment (mmusich) [inline Configuration/StandardSequences/python/AlCaRecoStream_Specials_cff.py]: @kodolova could you also please change the name of the responsible? Thanks

**Comment by diguida** [comment]
@kodolova 
As we were saying in person, at a first glance we were thinking that the definition of `hcalDigis.InputLabel = 'hltHcalCalibrationRaw'` was not needed. Actually, it is needed by HCAL NZS local reco `hcalLocalRecoSequenceNZS`. Can you please check?

**Comment by diguida** [comment]
@kodolova 
Please remember that the PD is now called `TestEnabledEcalHcal` as DT RAW collections will no more be there. Please check with the HLT contacts.

**Comment by kodolova** [comment]
I have checked. hcallocalNZSSequence introduces hbherecoMB = RecoLocalCalo.HcalRecProducers.HcalSimpleReconstructor_hbhe_cfi.hbheprereco.clone()
in ALCARECOHcalCalPedestal this instance get hcalDigiAlCaMB as digi
hbherecoMB.digiLabel = 'hcalDigiAlCaMB' which in turn get 'hltHcalCalibrationRaw as input
hcalDigiAlCaMB.InputLabel = 'hltHcalCalibrationRaw'

File: Configuration/StandardSequences/python/AlCaRecoStream_Specials_cff.py (hunk 2 above)
Comment (kodolova) [inline Configuration/StandardSequences/python/AlCaRecoStream_Specials_cff.py]: done

**Comment by diguida** [comment]
please test

**Comment by degano** [comment]
please test

**Comment by diguida** [comment]
@degano it seems that the failures are unrelated to this PR.

**Comment by diguida** [comment]
please test

**Comment by slava77** [comment]
-1 

The error was in the configuration, e.g. in 4.53:
```
ValueError: Trying to override definition of hcalDigis while it is used by the sequence RawToDigi
 new object defined in: /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_5_X_2015-03-24-2300/python/EventFilter/HcalRawToDigi/HcalRawToDigi_cfi.py
 existing object defined in: /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_5_X_2015-03-24-2300/python/Configuration/StandardSequences/RawToDigi_cff.py

```

IB itself didn't have problems

**Comment by kodolova** [comment]
I am not sure what to do with hcalDigis. This is the special producer which will be run only over TestEnabledEcalHcal: orbit gap data. Could you advice?

**Comment by davidlange6** [comment]
you need to clone the module and then modify it rather than modifying the existing one.

**Comment by kodolova** [comment]
That what was done. The line
#hcalDigis.InputLabel = 'hltHcalCalibrationRaw' was commented 4 days ago. 
There is a check of changes by Salvatore 4 days ago and this line should be commented.

File: Calibration/HcalAlCaRecoProducers/python/ALCARECOHcalCalPedestal_cff.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import FWCore.ParameterSet.Config as cms
+
+#------------------------------------------------
+#AlCaReco filtering for HCAL minbias:
+#------------------------------------------------
+
+from Calibration.HcalAlCaRecoProducers.ALCARECOHcalCalMinBias_cff import *
+hcalDigiAlCaMB.InputLabel = 'hltHcalCalibrationRaw'
+from EventFilter.HcalRawToDigi.HcalRawToDigi_cfi import *
```
Comment (slava77) [inline Calibration/HcalAlCaRecoProducers/python/ALCARECOHcalCalPedestal_cff.py]: is this import even needed?
all these can be tested locally with ``` runTheMatrix.py -s --useInput all ```

Comment (kodolova) [inline Calibration/HcalAlCaRecoProducers/python/ALCARECOHcalCalPedestal_cff.py]: 1. The main issue is that hcalDigis for TestEnabledEcalHcal get not-standard collection of digis:hltHcalCalibrationRaw
hcalDigis are defined in EventFilter.HcalRawToDigi.HcalRawToDigi_cfi as a
hcalDigis =  EventFilter.HcalRawToDigi.hcalRawToDigi_cfi.hcalRawToDigi.clone()
2. In the script the solution is:
from Calibration.HcalAlCaRecoProducers.ALCARECOHcalCalMinBias_cff import *
i.e. get hcalDigiAlCaMB = EventFilter.HcalRawToDigi.HcalRawToDigi_cfi.hcalDigis.clone(), change input
and then propagate this collect to NZS reconstruction.
I can take hcalDigis from Configuration/StandardSequences/python/RawToDigi_cff.py in ALCARECOHcalCalMinBias_cff and it should solve problem...

**Comment by mmusich** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
@cmsbuild please test
there were a couple of DAS related errors, most likely they blocked the completion of jenkins tests

**Comment by slava77** [comment]
@cmsbuild please test
[the success rate in jenkins is not very high these days]

**Comment by slava77** [comment]
@cmsbuild please test
[I couldn't find an area for this PR tests in jenkins, based on the request yesterday; I found only the failed one from 4 days ago; the host went offline there based on the logs https://cmssdt.cern.ch/jenkins/job/ib-any-integration/4258/console
]

**Comment by slava77** [comment]
@cmsbuild @nclopezo 
David, could you please check if jenkins tests are actually in order for this PR.
The bot is not sending "tests are being triggered" message anymore and the last attempt that I found in jenkins had a failure.

**Comment by nclopezo** [comment]
The tests are running now.

**Comment by degano** [comment]
@slava77 The issue was the bot started the tests (5 days ago) but never reported, thus assuming the tests where still running while it should have understood they probably crashed horribly.
We'll fix it.

**Comment by cvuosalo** [comment]
+1

For #8350 f22b5a292a5271199d14ca7422287835fe90699c

For HCAL, added configuration for NZS AlcaReco producer for the TestEnabledEcalHcalDT dataset.

The very small Reco code changes are satisfactory, and Jenkins [tests](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-8350/4310/summary.html) against baseline CMSSW_7_5_X_2015-04-20-2300 show no significant differences.

**Comment by mmusich** [comment]
+1

**Pull request merged**